### PR TITLE
implementing new push syntax

### PR DESCRIPTION
**Pull request by stackoverfloweth**
https://github.com/kitbagjs/router/assets/6098901/a6b43d33-8c6a-4b85-b986-e497dfebc565

### CODE REVIEW

File: src/utilities/hooks.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export async function runBeforeRouteHooks({ to, from, hooks }: BeforeContext): P
     if (error instanceof RouterPushError) {
       return {
         status: 'PUSH',
-        to: error.to,
+        to: error.to as Parameters<RegisteredRouterPush>,
```
Comment (pleek91) [inline src/utilities/hooks.ts:46]: ah yep you ran into the same issue here that I ran into in https://github.com/kitbagjs/router/pull/117

File: src/utilities/createRouterResolve.ts [hunk 2]
```diff
@@ -1,46 +1,50 @@
...
 
-export type RouterResolve<
-  TRoutes extends RouterRoutes
-> = <
-  TRoutePath extends string
->(source: string | RouteWithParams<TRoutes, TRoutePath>, options?: RouterResolveOptions) => string
+type ResolveRouteWithParamsArgs<TRoutes extends RouterRoutes, TRouteKey extends string> = [...RouteWithParamsArgs<TRoutes, TRouteKey>, options?: RouterResolveOptions | undefined]
 
-export type RouterResolveImplementation = (source: string | RouteWithParamsImplementation, options?: RouterResolveOptions) => string
+export type RouterResolve<TRoutes extends RouterRoutes> = {
```
Comment (pleek91) [inline src/utilities/createRouterResolve.ts:12]: this seems like it got a lot more complex

File: src/utilities/createRouterFind.ts [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
 import { getResolvedRouteForUrl } from '@/utilities/getResolvedRouteForUrl'
 
-export type RouterFind<
-  TRoutes extends RouterRoutes
-> = <
-  TRoutePath extends string
->(source: string | RouteWithParams<TRoutes, TRoutePath>) => ResolvedRoute | undefined
-
-export type RouterFindImplementation = (source: string | RouteWithParamsImplementation) => ResolvedRoute | undefined
+export type RouterFind<TRoutes extends RouterRoutes> = {
```
Comment (pleek91) [inline src/utilities/createRouterFind.ts:9]: the complexity 😢

**Comment by stackoverfloweth** [comment]
replaced by https://github.com/kitbagjs/router/pull/118

**Pull request closed without merging**